### ISSUE TITLE
Wrong multiple function call detection

### ISSUE THREAD

**Issue by vinhlh** [squizlabs/PHP_CodeSniffer#1114]
With this case
```php
<?php

class A
{
    public function test()
    {
        $this->setData(array(
                Webservice_Service_Constant::RESPONSE_KEY_SUCCESS => true,
                Webservice_Service_Constant::RESPONSE_KEY_MESSAGE => $inventoryStockConsignment->toArray())
        );
    }
}
```
it is detected as single line.

There is a fix
```php
--- a/CodeSniffer/Standards/PSR2/Sniffs/Methods/FunctionCallSignatureSniff.php
+++ b/CodeSniffer/Standards/PSR2/Sniffs/Methods/FunctionCallSignatureSniff.php
@@ -51,8 +51,12 @@ class PSR2_Sniffs_Methods_FunctionCallSignatureSniff extends PEAR_Sniffs_Functio
     {
         // If the first argument is on a new line, this is a multi-line
         // function call, even if there is only one argument.
+        //
+        // if the first argument is an array having the close parenthesis is on the next line
         $next = $phpcsFile->findNext(PHP_CodeSniffer_Tokens::$emptyTokens, ($openBracket + 1), null, true);
-        if ($tokens[$next]['line'] !== $tokens[$stackPtr]['line']) {
+        if ($tokens[$next]['line'] !== $tokens[$stackPtr]['line']
+            || ($tokens[$next]['type'] === 'T_ARRAY'
+                && $tokens[$tokens[$next]['parenthesis_closer']]['line'] !== $tokens[$stackPtr]['line'])) {
             return true;
         }
```

**Comment by aik099** [comment]
What are the errors you're getting from running `phpcs -s` (the `-s` shows sniff codes in error message)?

**Comment by vinhlh** [comment]
@aik099 
```
 7 | ERROR | [x] Expected 0 spaces before closing bracket; newline
   |       |     found
   |       |     (PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket)
```

**Comment by gsherwood** [comment]
You've got the same problem as was reported on issue https://github.com/squizlabs/PHP_CodeSniffer/pull/1093

Namely, that this function call is actually classified as single-line by PSR2, as clarified by the PHP-FIG and included in the PSR2 errata. The errata actually uses an array as an example of what is allowed.

Go have a read of my comment on that other issue and hopefully that will explain things: https://github.com/squizlabs/PHP_CodeSniffer/pull/1093#issuecomment-236120017

Does it make sense after reading that comment and the links?

**Comment by vinhlh** [comment]
@gsherwood it makes sense, thank you.
